Add assert_tensor_equal and assert_tensor_not_equal to test/cpp/api/support.h

yf225 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #29847 C++ tensor indexing: add Tensor::idx() and Tensor::idx_put_(), and simple indexing tests
* **#29846 Add assert_tensor_equal and assert_tensor_not_equal to test/cpp/api/support.h**
* #29845 C++ tensor indexing: add Slice / TensorIndex

This PR adds `assert_tensor_equal` and `assert_tensor_not_equal` to `test/cpp/api/support.h`, as better functions for testing whether two tensors are equal / not equal.

Differential Revision: [D18694889](https://our.internmc.facebook.com/intern/diff/D18694889)

Commit:
Add assert_tensor_equal and assert_tensor_not_equal to test/cpp/api/support.h

Commit:
Update on "Add assert_tensor_equal and assert_tensor_not_equal to test/cpp/api/support.h"

Add assert_tensor_equal and assert_tensor_not_equal to test/cpp/api/support.h

gh-metadata: pytorch pytorch 29846 gh/yf225/73/head

kostmo commented:
## CircleCI build failures summary
As of commit 09beee9:
* **3/3** failures introduced in this PR
* **0/3** recognized as flaky



Here are the [reasons each build failed](https://dr.pytorch.org/commit-details.html?sha1=09beee96081f0b8c7b0f2ceb4dbe78dd64b1cc6c):

| Job | Step | Log excerpt |
| --- | --- | --- |
| caffe2_onnx_py3_6_clang7_ubuntu16_04_build | Build | `/var/lib/jenkins/workspace/test/cpp/api/support.h:111:10: error:` |
| caffe2_onnx_py2_gcc5_ubuntu16_04_build | Build | `/var/lib/jenkins/workspace/test/cpp/api/support.h:111:10: error:` |


---
This comment was automatically generated by [Dr. CI](https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment).
Follow [this link to opt-out](https://dr.pytorch.org/admin/comments-opt-out.html) of these comments for your Pull Requests.


Please report bugs/suggestions on the [GitHub issue tracker](https://github.com/kostmo/circleci-failure-tracker/issues).

Commit:
Update on "Add assert_tensor_equal and assert_tensor_not_equal to test/cpp/api/support.h"

Add assert_tensor_equal and assert_tensor_not_equal to test/cpp/api/support.h

gh-metadata: pytorch pytorch 29846 gh/yf225/73/head

Commit:
Update on "Add assert_tensor_equal and assert_tensor_not_equal to test/cpp/api/support.h"

Add assert_tensor_equal and assert_tensor_not_equal to test/cpp/api/support.h

gh-metadata: pytorch pytorch 29846 gh/yf225/73/head

Commit:
Update on "Add assert_tensor_equal and assert_tensor_not_equal to test/cpp/api/support.h"

Add assert_tensor_equal and assert_tensor_not_equal to test/cpp/api/support.h

gh-metadata: pytorch pytorch 29846 gh/yf225/73/head

Commit:
Update on "Add assert_tensor_equal and assert_tensor_not_equal to test/cpp/api/support.h"

Add assert_tensor_equal and assert_tensor_not_equal to test/cpp/api/support.h

gh-metadata: pytorch pytorch 29846 gh/yf225/73/head

Commit:
Update on "Add assert_tensor_equal and assert_tensor_not_equal to test/cpp/api/support.h"

Add assert_tensor_equal and assert_tensor_not_equal to test/cpp/api/support.h

gh-metadata: pytorch pytorch 29846 gh/yf225/73/head

Commit:
Update on "Add assert_tensor_equal and assert_tensor_not_equal to test/cpp/api/support.h"

Add assert_tensor_equal and assert_tensor_not_equal to test/cpp/api/support.h

gh-metadata: pytorch pytorch 29846 gh/yf225/73/head

yf225 commented:
Closing in favor of https://github.com/pytorch/pytorch/pull/30426.
